### PR TITLE
Crossplatform ByteString implementation

### PR DESCRIPTION
**Pull request by Egorand**
Prerequisite for #431

### CODE REVIEW

File: okio/src/main/kotlin/okio/common/ByteString.kt [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+package okio.common
+
+import okio.ByteString
+import okio.and
+import okio.arrayRangeEquals
+import okio.arraycopy
+import okio.hashCode
+
+// TODO Kotlin's expect classes can't have default implementations, so platform implementations
```
[the remark this answers is not in the document]
Comment (Egorand) [reply okio/src/main/kotlin/okio/common/ByteString.kt]: This won't work for `Serializable` private methods which have to be inside the class to work

File: okio/src/main/kotlin/okio/ByteString.kt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  fun utf8(): String
+
+  /**
+   * Returns this byte string encoded as [Base64](http://www.ietf.org/rfc/rfc2045.txt). In violation
+   * of the RFC, the returned string does not wrap lines at 76 columns.
+   */
+  fun base64(): String
+
+  /** Returns the 128-bit MD5 hash of this byte string.  */
+  fun md5(): ByteString
```
[the remark this answers is not in the document]
Comment (Egorand) [reply okio/src/main/kotlin/okio/ByteString.kt]: Implementation relies on Java classes, so not completely

[the remark this answers is not in the document]
Comment (swankjesse) [reply okio/src/main/kotlin/okio/ByteString.kt]: Omit crypto functions for now?

File: okio/src/main/kotlin/okio/platform.kt [hunk 3]
```diff
@@ -37,9 +39,14 @@ expect annotation class JvmStatic()
...
 // TODO make internal https://youtrack.jetbrains.com/issue/KT-19664
 // TODO or remove after https://youtrack.jetbrains.com/issue/KT-18882
-@Target(FILE, FUNCTION)
+@Target(FILE, FUNCTION, PROPERTY_GETTER, PROPERTY_SETTER)
 expect annotation class JvmName(val name: String)
 
+// TODO make internal https://youtrack.jetbrains.com/issue/KT-19664
+// TODO or remove after https://youtrack.jetbrains.com/issue/KT-18882
+@Target(FIELD)
+expect annotation class Transient()
```
Comment (swankjesse) [inline okio/src/main/kotlin/okio/platform.kt]: Possible to define this in the JVM class only?

Comment (Egorand) [reply okio/src/main/kotlin/okio/platform.kt]: aed0d0c

**Pull request merged**